src/controllers/product.controller.js

```diff
@@ -37,3 +44,29 @@ export const getAllProducts = async (req, res) => {
       .json({ error: err.message, message: 'Failed to retrieve products' });
   }
 };
+export const getSpecificProduct = async (req, res) => {
+  try {
+    const { productId } = req.params;
+    const { role, id } = req.user;
+    const product = await ProductService.getSingleProduct(productId);
+    if (product) {
+      if (role === 'seller') {
```

Comment:
This condition should also have a condition to check if userId  == product.userId